src/main/webapp/app/programming/manage/code-editor/instructor-and-editor-container/code-editor-instructor-and-editor-container.component.ts

```diff
@@ -48,6 +54,62 @@ export class CodeEditorInstructorAndEditorContainerComponent extends CodeEditorI
     faTimes = faTimes;
     faCircleNotch = faCircleNotch;
     faTimesCircle = faTimesCircle;
+    faSpinner = faSpinner;
+    facArtemisIntelligence = facArtemisIntelligence;
     irisSettings?: IrisSettings;
     protected readonly RepositoryType = RepositoryType;
+
+    private codeGenerationService = inject(HyperionCodeGenerationService);
+    private codeGenAlertService = inject(AlertService);
+    isGeneratingCode = false;
```

Comment:
Here you could/should also use the new signal API
```suggestion
    isGeneratingCode = signal(false);
```

Reply:
It helps with reactive state management and will be used in the future when we go zoneless (remove Zone.js as rendering mode for Angular)